### DISCUSSION TITLE
confused about lamda-bind

### DISCUSSION THREAD

**Discussion by DanielAjoy** [h3rald/min#179] [General]
The documentation of **lambda-bind** says

> Binds the specified quotation to an existing symbol 'sym which was previously-set via lambda.

But it seems that I can do just that with **lambda** itself. Here I believe I'm binding the quotation "(2)" to the existing symbol **uno** which was previously-set via **lambda**.

```
$ (1) 'uno lambda (2) 'uno lambda uno
 2
```

Here, I don't see a difference between lambda and lambda-bind.

**Comment by h3rald** [comment]
Well, agreed, in that case there's no difference, because the symbol is defined in the same scope.

If you were in an inner scope, calling `lambda` would create a new symbol local to that scope, while `lambda-bind` re-binds an existing symbol which may also be in an outer scope. It's basically the same thing as `define` vs. `bind`, I had to differentiate it because otherwise you would have to double-quote quotations to bind them "as data" (i.e. arrays) to avoid being executed when placed on the stack.
